recipes/default.rb

```diff
@@ -20,7 +20,7 @@
 # Installs base package, creates log directory, and write out CDAP configuration
 include_recipe 'cdap::base'
 
-if node['cdap'].key?('cdap_env') && node['cdap']['cdap_env'].key?('log_dir')
+if cdap_property('log_dir', 'cdap_env')
```

Comment:
original comment applied to when ``cdap_property`` returned the value.  The guard is still redundant, and should use the helper instead of duplicating the helper code